.github/workflows/frontend.yaml

```diff
@@ -0,0 +1,71 @@
+name: Frontend
+on:
+  push:
+    branches: [main]
+  pull_request:
+    paths:
+      - frontend/**
+jobs:
+  build:
+    name: Build, test & lint
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: frontend
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+      - name: Get NPM cache folder path
+        id: npm-cache-folder
+        run: echo "::set-output name=path::$(npm config get cache)"
+      - name: Set up NPM cache
+        uses: actions/cache@v3
+        id: npm-cache
+        env:
+          cache-name: frontend-node-modules
+        with:
+          path: ${{ steps.npm-cache-folder.outputs.path }}
+          key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.cache-name }}-
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - run: ls
+      - name: Run unit tests
+        run: npm run test:unit
+      - name: Run end-to-end tests
+        run: npm run test:e2e:ci
+      - name: Lint
```

Comment:
Poustel bych lint pred buildem. Pripadne pro rychlejsi a lepsi feedback pro commitujici vyvojare jeste pouzivame paralelni beh lintu a testu, je to na uvazeni.

Reply:
OK, může být. Zatím bych to nerozděloval, kdyžtak pokud uvidíme, že se to tím nějak zásadně zpomaluje.